Support torchbind as attribute in torch.fx symbolic tracing

houseroad opened the pull request:
Summary: add support for ScriptObject as attributes in symbolic trace.

Test Plan: OSS CI

Differential Revision: D25116185

jamesr66a commented on test/test_fx.py:
```diff
@@ -1139,6 +1139,23 @@ def forward(self):
         self.checkGraphModule(m, ())
 
+    def test_torchbind_class_attribute_in_fx(self):
+        try:
```
You can replace this try/catch with this incantation:

https://github.com/pytorch/pytorch/blob/fadec77c301d21e04cbc1d9652f2cf29dc4e7572/test/test_fx.py#L300

jamesr66a commented on torch/fx/symbolic_trace.py:
```diff
@@ -2,6 +2,7 @@
 from types import CodeType, FunctionType
 from typing import Any, Dict, Optional, List, Callable, Union
 import torch
+from torch._C import ScriptObject  # type: ignore
```
What was the lint error here?

houseroad replied:
it's Error: Module 'torch._C' has no attribute 'ScriptModule'  [attr-defined]

Tried to add it in the mypy.ini for ignore import error, but it didn't work. A little bit hesitate to mark it ignore all the errors. So just mark this linne.

houseroad committed:
Support torchbind as attribute in torch.fx symbolic tracing (#48732)

Summary:
Pull Request resolved: https://github.com/pytorch/pytorch/pull/48732

add support for ScriptObject as attributes in symbolic trace.

Test Plan: OSS CI

Reviewed By: jamesr66a

Differential Revision: D25116185

fbshipit-source-id: 24c18457934fb78c81800dee6f5dbf1349fcdaf1
